insomniac.py

```diff
@@ -169,6 +182,58 @@ def on_likes_limit_reached():
                 raise e
 
 
+def _job_handle_hashtags(device, hashtags, likes_count, follow_percentage, storage, profile_filter, on_interaction):
+    class State:
+        def __init__(self):
+            pass
+
+        is_job_completed = False
+
+    state = State()
+    session_state = sessions[-1]
+
+    def on_likes_limit_reached():
+        state.is_job_completed = True
+
+    on_interaction = partial(on_interaction, on_likes_limit_reached=on_likes_limit_reached)
+
+    for hashtag in hashtags:
+        
+        print_timeless("")
+        print(COLOR_BOLD + "Handle #" + hashtag + (is_myself and " (it\'s you)" or "") + COLOR_ENDC)"""
+        completed = False
+        
+        while not completed and not state.is_job_completed:
+            try:
```

Comment:
I've recently made a decorator `@_run_safely` that incapsulates all this try..except block to avoid code duplication. You can now merge master into this branch and simply replace this whole block with

```
@_run_safely(device=device)
def job():
    handle_hashtags(device, 
                                 hashtag, 
                                 likes_count, 
                                 follow_percentage, 
                                 storage,
                                 profile_filter,  
                                 _on_like, 
                                 on_interaction)
    state.is_job_completed = True

while not state.is_job_completed:
    job()
```